### PR TITLE
Fixed time coordinate of MIRCO-ESM

### LINKED ISSUE #1187: Dataset problem: invalid time bounds of `gpp_Lmon_MIROC-ESM_esmFixClim1_r1i1p1_000101-014012.nc`
**Describe the dataset issue**

The first time bound of `gpp_Lmon_MIROC-ESM_esmFixClim1_r1i1p1_000101-014012.nc` is below `datetime.MINYEAR`, which causes trouble when `unit.num2date` is used.

### PR DESCRIPTION
**Pull request by schlunma**
Fix time coordinate of `gpp_Lmon_MIROC-ESM_esmFixClim1_r1i1p1_000101-014012.nc`.

Closes #1187.

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/cmor/_fixes/cmip5/miroc_esm.py [hunk 1]
```diff
@@ -91,4 +94,18 @@ def fix_metadata(self, cubes):
...
 
+            # Fix time
+            if cube.coords('time'):
+                expected_time_units = Unit('days since 1950-1-1 00:00:00',
+                                           calendar='gregorian')
+                if cube.coord('time').units != expected_time_units:
+                    continue
+                if cube.coord('time').bounds is None:
+                    continue
+                if np.isclose(cube.coord('time').bounds[0][0], -711860.5):
```
Comment (valeriupredoi) [inline esmvalcore/cmor/_fixes/cmip5/miroc_esm.py:105]: could you maybe write a comment above the if statement explaining what -711860.5 means? Future Manu in 5 years will be scratching his head :grin:

File: tests/integration/cmor/_fixes/cmip5/test_miroc_esm.py [hunk 2]
```diff
@@ -79,6 +80,19 @@ def setUp(self):
                                 calendar='gregorian')), 0)
         self.cube.add_dim_coord(DimCoord([0, 1], long_name='AR5PL35'), 1)
 
+        time_units = Unit('days since 1950-1-1 00:00:00', calendar='gregorian')
+        time_coord = DimCoord(
+            [-711845.0, -711814.0],
+            bounds=[[-711860.5, -711829.5], [-711829.5, -711800.0]],
+            var_name='time',
```
Comment (valeriupredoi) [inline tests/integration/cmor/_fixes/cmip5/test_miroc_esm.py:87]: same here, a comment saying those numbers are 0 days from 1950-1-1 (?)

**Review by valeriupredoi — APPROVED** [review]
just a wee comment needed say what those numbers are, all else good by me, cheers Manu :+1:

**Comment by schlunma** [comment]
Thanks for the review @valeriupredoi! Please merge when you are happy with the comments

**Pull request merged**